retrace-server.spec.in

```diff
@@ -99,7 +100,42 @@ rm -f %{buildroot}%{_infodir}/dir
 #retrace uid/gid reserved in setup, rhbz #706012
 %define retrace_gid_uid 174
 getent group retrace > /dev/null || groupadd -f -g %{retrace_gid_uid} --system retrace
-getent passwd retrace > /dev/null || useradd --system -g retrace -u %{retrace_gid_uid} -d %{_datadir}/%{name} -s /sbin/nologin retrace
+getent passwd retrace > /dev/null || useradd --system -g retrace -u %{retrace_gid_uid} -b %{_sharedstatedir} -s /sbin/nologin retrace
```

Comment:
I encourted a problem with this. I already have a retrace user on my system (from older installation) with a home dir `/usr/share/retrace`.

Upgrading retrace-server does not change the home dir to `/var/lib/retrace`.

https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/

Reply:
Seems to be working now. It's quite ugly because simple `usermod -m -d` will fail if there is a process run by the user `retrace`. So I just `userdel -f` the user and then add it again with a different home dir.  
  
Side note for @ernestask: `userdel` removes the appropriate entries from `/etc/sub[ug]id` but it leaves the empty lines in place, meaning that the files did end up slightly malformed. After many testing installs and uninstalls they contained one entry for my user followed by many `\n`'s. Fortunately, `sort -n` puts the blank lines at the top but I added a `sed` call to get rid of those anyway.

Reply:
In cut(1):
```
       -s, --only-delimited
              do not print lines not containing delimiters
```

That should do the trick as well.

Reply:
By “malformed” I meant anything that isn’t empty or in the right format. Probably should have clarified.

Reply:
I dropped this whole part from the .spec file altogether and just documented it instead.